src/main/java/io/jenkins/plugins/coverage/CoverageChecksPublisher.java

```diff
@@ -0,0 +1,104 @@
+package io.jenkins.plugins.coverage;
+
+import hudson.model.Run;
+import io.jenkins.plugins.checks.api.*;
+import io.jenkins.plugins.checks.api.ChecksOutput.ChecksOutputBuilder;
+import io.jenkins.plugins.checks.api.ChecksDetails.ChecksDetailsBuilder;
+import io.jenkins.plugins.coverage.targets.*;
+
+import java.util.*;
+
+/**
+ * Publishes coverage as checks to scm platforms.
+ *
+ * @author Kezhi Xiong
+ */
+class CoverageChecksPublisher {
+    private final CoverageAction action;
+
+    CoverageChecksPublisher(final CoverageAction action) {
+        this.action = action;
+    }
+
+    void publishChecks() {
+        ChecksPublisher publisher = ChecksPublisherFactory.fromRun(action.getOwner());
+        publisher.publish(extractChecksDetails());
+    }
+
+    private ChecksDetails extractChecksDetails() {
+        CoverageResult result = action.getResult();
+        ChecksOutputBuilder outputBuilder = new ChecksOutputBuilder()
+                .withTitle("Code Coverage")
+                .withSummary("## Generated " + result.getResults().size() + " types of coverage.")
+                .withText(extractChecksText(result)); // TODO: improve summary
+
+        return new ChecksDetailsBuilder()
+                .withName("Code Coverage")
```

Comment:
not sure which field it is as they have the same text currently but can a summary go on the check status on the right hand side

![image](https://user-images.githubusercontent.com/21194782/87909135-8e84e780-ca5f-11ea-938e-1e1e3cfbfd24.png)

Reply:
`title` field in the `output` object. Then I'll change that part in warnings checks as well.

Reply:
take a look at this plugin: https://github.com/jenkinsci/github-coverage-reporter-plugin